src/hooks/useProductById.ts

```diff
@@ -8,31 +8,42 @@ const useMockData = import.meta.env.USE_MOCK_FOR_API_FAIL === 'true';
 interface UseProductByIdResult {
   product: ProductType | null;
   isLoading: boolean;
+  status?: number; // Optional status code
 }
 
-const fetchProductById = async (productId: string) => {
+const fetchProductById = async (productId: string, useMockData: boolean) => {
   if (!productId) {
-    return null;
+    return { data: null, status: 0 };
```

Comment:
`status: 404` instead you can read more at https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
Fixed